src/main/java/com/teamc/fms/service/TimeSlotService.java

```diff
@@ -0,0 +1,22 @@
+package com.teamc.fms.service;
+
+import com.teamc.fms.dto.TimeSlotDto;
+import com.teamc.fms.entity.TimeSlot;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface TimeSlotService {
+
+    TimeSlotDto updateTimeSlot(String id, TimeSlotDto timeSlot);
+
+    //    List<TimeSlot> getTimeSlotsOrderByCodeAsc();
+    List<TimeSlot> getAllTimeSlots();
+
+    TimeSlotDto addTimeSlot(TimeSlotDto timeSlot);
```

Comment:
As agreed, we should return an entity and then map it in the controller
We should receive a record instead of (TimeSlotDto timeSlot), the record will contain all the required data to create the timeslot  (CreateTimeslotCommand)
mapping from createTimeslotDto to CreateTimeslotCommand is done in controller